### PR TITLE
Additions to `additional_minor_features.md`

### PR DESCRIPTION
**Pull request by halirutan**

### CODE REVIEW

**Comment by halirutan** [comment]
@i582 Not a problem. I'll just remove my section. Le me ping Karol so that we get your PR through first.

File: topics/reference_guide/custom_language_support/additional_minor_features.md [hunk 1]
```diff
@@ -121,6 +121,17 @@ EP: `com.intellij.include.provider`
...
 Including/included files can then be obtained via [`FileIncludeManager`](upsource:///platform/lang-api/src/com/intellij/psi/impl/include/FileIncludeManager.java).
 
+### Recognizing Complex Multi-Block Expressions
+
+EP: `com.intellij.codeBlockSupportHandler`
+
+[`CodeBlockSupportHandler`](upsource:///platform/lang-impl/src/com/intellij/codeInsight/highlighting/CodeBlockSupportHandler.java)
+allows providing text ranges for more complex code blocks like, e.g., in `if`/`elsif`/`else` blocks.
+It is used to highlight markers and keywords if one is under the cursor, and for navigation to the beginning/end of blocks.
+See also the [paired brace matching](additional_minor_features.md#brace-matching).
```
Comment (YannCebron) [inline topics/reference_guide/custom_language_support/additional_minor_features.md:131]: add reverse link there as well, can the link be shortened to just `(#header)`?

File: topics/reference_guide/custom_language_support/additional_minor_features.md [hunk 2]
```diff
@@ -165,6 +165,14 @@ EP: `completion.plainTextSymbol`
 provides a simple way to extract lookup elements from a file so that users have completion available
 in, e.g., plain text editors like VCS commit messages.
 
+### Splitting and Joining List Constructs
+
+EP: `com.intellij.listSplitJoinContext`
+
+[`ListSplitJoinContext`](upsource:///platform/lang-impl/src/com/intellij/openapi/editor/actions/lists/ListSplitJoinContext.kt)
```
Comment (YannCebron) [inline topics/reference_guide/custom_language_support/additional_minor_features.md:172]: clarify how to invoke this action in editor/UI

File: topics/reference_guide/custom_language_support/additional_minor_features.md [hunk 3]
```diff
@@ -173,6 +173,16 @@ EP: `com.intellij.listSplitJoinContext`
 needs to be implemented to define the exact behavior of splitting and joining list-like constructs
 in a language.
 
+### Rename and Change Signature Refactoring
+
+EP: `com.intellij.suggestedRefactoringSupport`
+
+[`SuggestedRefactoringSupport`](upsource:///platform/lang-api/src/com/intellij/refactoring/suggested/SuggestedRefactoringSupport.kt)
+provides functionality to implement rename and change signature refactorings for custom languages.
```
Comment (YannCebron) [inline topics/reference_guide/custom_language_support/additional_minor_features.md:181]: the key is to automatically _suggest_ such refactorings by detecting such changes

File: topics/reference_guide/custom_language_support/additional_minor_features.md [hunk 4]
```diff
@@ -193,6 +193,23 @@ Please see
...
+
+[`EditorTabColorProvider`](upsource:///platform/ide-core-impl/src/com/intellij/openapi/fileEditor/impl/EditorTabColorProvider.java)
+allows for the modification of the background colors for specific files.
+
+### Custom Names and Tooltips for Editor Tabs
+
+EP: `com.intellij.editorTabTitleProvider`
+
+[`EditorTabTitleProvider`](upsource:///platform/ide-core-impl/src/com/intellij/openapi/fileEditor/impl/EditorTabTitleProvider.java)
+allows for specifying custom names and tooltips displayed in the titel of editor tabs.
```
Comment (YannCebron) [inline topics/reference_guide/custom_language_support/additional_minor_features.md:208]: titel -> title

File: topics/reference_guide/custom_language_support/additional_minor_features.md (hunk 1 above)
Comment (halirutan) [reply topics/reference_guide/custom_language_support/additional_minor_features.md:131]: Fixed and yes, you can just use `#header` it seems.

File: topics/reference_guide/custom_language_support/additional_minor_features.md (hunk 3 above)
Comment (halirutan) [reply topics/reference_guide/custom_language_support/additional_minor_features.md:181]: Fixed

File: topics/reference_guide/custom_language_support/additional_minor_features.md (hunk 2 above)
Comment (halirutan) [reply topics/reference_guide/custom_language_support/additional_minor_features.md:172]: Improved drastically

File: topics/reference_guide/custom_language_support/additional_minor_features.md (hunk 4 above)
Comment (halirutan) [reply topics/reference_guide/custom_language_support/additional_minor_features.md]: Fixed

**Review by karollewandowski — APPROVED** [review]

File: topics/reference_guide/custom_language_support/additional_minor_features.md [hunk 5]
```diff
@@ -28,6 +28,7 @@ The IDE will match such braces, but the highlighting for such braces will not be
 Certain types of braces can be marked as structural.
 Structural braces have higher priority than regular braces: they are matched with each other even if there are unmatched braces of different types between them.
 An opening non-structural brace is not matched with a closing one if one of them is inside a pair of matched structural braces and another is outside.
+See also the [Recognizing Complex Multi-Block Expressions](#recognizing-complex-multi-block-expressions) item.
```
Comment (YannCebron) [inline topics/reference_guide/custom_language_support/additional_minor_features.md:31]: should be possible to leave out the link title

File: topics/reference_guide/custom_language_support/additional_minor_features.md [hunk 6]
```diff
@@ -128,7 +129,7 @@ EP: `com.intellij.codeBlockSupportHandler`
 [`CodeBlockSupportHandler`](upsource:///platform/lang-impl/src/com/intellij/codeInsight/highlighting/CodeBlockSupportHandler.java)
 allows providing text ranges for more complex code blocks like, e.g., in `if`/`elsif`/`else` blocks.
 It is used to highlight markers and keywords if one is under the cursor, and for navigation to the beginning/end of blocks.
-See also the [paired brace matching](additional_minor_features.md#brace-matching).
+See also the [paired brace matching](#brace-matching) item.
```
Comment (YannCebron) [inline topics/reference_guide/custom_language_support/additional_minor_features.md:132]: link title = remove

**Pull request merged**